src/core/types.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * @typedef {object} AlloyMonitor
+ * @property {function(data): void} [onBeforeLog]
+ * @property {function(data): void} [onInstanceCreated]
+ * @property {function(data): void} [onInstanceConfigured]
+ * @property {function(data): void} [onBeforeCommand]
+ * @property {function(data): void} [onCommandResolved]
+ * @property {function(data): void} [onCommandRejected]
+ * @property {function(data): void} [onBeforeNetworkRequest]
+ * @property {function(data): void} [onNetworkResponse]
+ * @property {function(data): void} [onNetworkError]
+ * @property {function(data): void} [onContentHiding]
+ * @property {function(data): void} [onContentRendering]
+ */
+
+export const Types = {};
```

Comment:
Why have an export here?

Reply:
It's a convention to make the file play nice with the module system and the tools that read JSDoc comments for type information.